semtypes/src/main/java/io/ballerina/types/typeops/FieldPair.java

```diff
@@ -28,14 +30,20 @@
     public final String  name;
     public final SemType type1;
     public final SemType type2;
+    Optional<Integer> index1;
```

Comment:
Can we use `Integer` with `null` here? It's not a common practice to have optional fields